invoices/invoiceregistry.go

```diff
@@ -698,8 +719,28 @@ func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx) error {
 	_, err = i.AddInvoice(invoice, ctx.hash)
 	if err != nil && err != channeldb.ErrDuplicateInvoice {
 		return err
+	} else if err == channeldb.ErrDuplicateInvoice {
+		dbInvoice, err := i.LookupInvoice(ctx.hash)
+		if err != nil {
+			return err
+		}
+
+		// Extract the payment address, which was generated at random on
+		// the initial pass.
+		payAddr = dbInvoice.Terms.PaymentAddr
```

Comment:
What about keysend instances that re-use the same payment hash, and didn't have a payment addr stored (were made before this PR)?

Reply:
as of #4285, we'll fallback to finding the invoice by payment hash if the payment addr fails